src/cpp/session/modules/SessionFind.cpp

```diff
@@ -301,26 +513,213 @@ class GrepOperation : public boost::enable_shared_from_this<GrepOperation>
          else
             pMatchOff->push_back(json::Value(gsl::narrow_cast<int>(nUtf8CharactersProcessed)));
       }
-      
+
       if (inputPos != end)
          decodedLine.append(decode(std::string(inputPos, end)));
 
+      *pFullLineContent = decodedLine;
```

Comment:
I added an else pFullLineContent = pContent; 
and it worked inside the function, but once we returned from processContents, lineInfo.decodedContents had the encoded  version. Is this what you were recommending?

Reply:
I'm not quite following what you mean by `lineInfo.decodedContents` had the encoded version. Do you have any idea why that would happen? Also if it's not possible to do `pFullLineContent = pContent`, is it possible to null out `pFullLineContent` and only look at `pContent` in that case?

Reply:
lineInfo.decodedContents is the variable we're passing for pFullLineContent. I debugged the code and confirmed pFullLineContent had the proper value after setting pContent = decodedLine. But outside of the function lineInfo.decodedContents did not have this value. At this point I'm not sure why. 
I originally wrote the code to null out pFullLineContent, but it just required to set the variable in lineInfo later on so I'm not sure what the benefit would be to doing it later on.

Reply:
Ah, that makes sense! It's because you're change the `pFullLineContent` pointer to contain a different address, rather than changing the contents at the location it originally pointed to ('doh!). If that value ends up needing to be set anyways, I think you're right that there's no point to changing this code.